src/librawspeed/decoders/DngDecoder.cpp

```diff
@@ -672,22 +674,25 @@ void DngDecoder::decodeMetaDataInternal(const CameraMetaData* meta) {
         mRaw->metadata.wbCoeffs[i] = (c > 0.0F) ? (1.0F / c) : 0.0F;
       }
     }
-  } else if (mRootIFD->hasEntryRecursive(TiffTag::ASSHOTWHITEXY)) {
+  } else if (!mRaw->metadata.colorMatrix.empty() &&
+             mRootIFD->hasEntryRecursive(TiffTag::ASSHOTWHITEXY)) {
     const TiffEntry* as_shot_white_xy =
         mRootIFD->getEntryRecursive(TiffTag::ASSHOTWHITEXY);
     if (as_shot_white_xy->count == 2) {
-      mRaw->metadata.wbCoeffs[0] = as_shot_white_xy->getFloat(0);
-      mRaw->metadata.wbCoeffs[1] = as_shot_white_xy->getFloat(1);
-      mRaw->metadata.wbCoeffs[2] =
-          1 - mRaw->metadata.wbCoeffs[0] - mRaw->metadata.wbCoeffs[1];
-
-      const std::array<float, 3> d65_white = {{0.950456, 1, 1.088754}};
-      for (uint32_t i = 0; i < 3; i++)
-        mRaw->metadata.wbCoeffs[i] /= d65_white[i];
+      const float x = as_shot_white_xy->getFloat(0);
+      const float y = as_shot_white_xy->getFloat(1);
+      const std::array<float, 3> as_shot_white = {{x / y, 1, (1 - x - y) / y}};
```

Comment:
Please add a link to http://www.brucelindbloom.com/index.html?Eqn_xyY_to_XYZ.html,
replacing hardcoded `1` with `constexpr float Y = 1.0;`

Reply:
Done